rootio/scanner.go

```diff
@@ -54,11 +54,36 @@ func (s *baseScanner) SeekEntry(i int64) error {
 	if s.err != nil {
 		return s.err
 	}
+	if s.chain {
+		if s.cur >= s.n {
+			itree := s.findTree(i)
+			s.loadTree(itree)
+		}
+	}
 	s.i = i
 	s.cur = i - 1
 	return s.err
 }
 
+// find the tree number in the chain that contains entry i
+func (s *baseScanner) findTree(i int64) int {
+	ch := s.tree.(*tchain)
+	var j int = 0
+	for j <= len(ch.trees) {
+		if i <= ch.tree.Entries() {
```

Comment:
I don't think this works.
`ch.tree` is always pointing at the same tree, so `ch.tree.Entries()` will always return the same value.

I believe the correct code would be:
```go
for j, tree := range ch.trees {
    if i < tree.Entries() {
         return j
    }
}
return -1
```